src/main/java/org/carlmontrobotics/subsystems/IntakeShooter.java

```diff
@@ -4,47 +4,120 @@
+import static org.carlmontrobotics.Constants.IntakeShooter.*;
+
+//import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+//import edu.wpi.first.networktables.NetworkTable;
+//import edu.wpi.first.networktables.NetworkTableEntry;
+//import edu.wpi.first.networktables.NetworkTableInstance;
+//these 4 imports above are for limelight
+
 
 
 public class IntakeShooter extends SubsystemBase {
-    CANSparkMax leftFiringMotor = MotorControllerFactory.createSparkMax(0, MotorConfig.NEO);
-    CANSparkMax rightFiringMotor = MotorControllerFactory.createSparkMax(1,MotorConfig.NEO);
-    CANSparkMax leftPassMotor = MotorControllerFactory.createSparkMax(2, MotorConfig.NEO_550);
-		CANSparkMax rightPassMotor = MotorControllerFactory.createSparkMax(3,MotorConfig.NEO_550);
-	
-		CANSparkMax intakeMotor = MotorControllerFactory.createSparkMax(4, MotorConfig.NEO_550);
-//	CANSparkMax rightIntakeMotor = MotorControllerFactory.createSparkMax(5, MotorConfig.NEO_550);
-	
-		public IntakeShooter() {
-    }
-	
-		public void intake() {}//run both intake and passing motors
-    public void fire() {}//run both passing and firing motors
+    CANSparkMax leftShootingMotor = MotorControllerFactory.createSparkMax(leftShootingPort, MotorConfig.NEO);
+    CANSparkMax rightShootingMotor = MotorControllerFactory.createSparkMax(rightShootingPort, MotorConfig.NEO);
+    CANSparkMax intakeMotor = MotorControllerFactory.createSparkMax(intakePort, MotorConfig.NEO_550);
+
+	private final RelativeEncoder leftShootingEncoder = leftShootingMotor.getEncoder();
+	private final RelativeEncoder rightShootingEncoder = rightShootingMotor.getEncoder();
+
+	private final RelativeEncoder intakeMotorEncoder = intakeMotor.getEncoder();
+	//todo: figure out way to make it so its not "constants."
+	private final SparkPIDController leftShootingMotorPID = leftShootingMotor.getPIDController();
+	private final SparkPIDController rightShootingMotorPID = rightShootingMotor.getPIDController();
+	private final TimeOfFlight distSensor = new TimeOfFlight(distSensorPort);
+
+	//private final DigitalOutput beamBreak = new DigitalOutput(beamBreakPort);
+	//We may use this
+
+		//NetworkTable table = NetworkTableInstance.getDefault().getTable("limelight");
+		//NetworkTableEntry tx = table.getEntry("tx");
+		//NetworkTableEntry ty = table.getEntry("ty");
+		//NetworkTableEntry ta = table.getEntry("ta");
+
+		//SmartDashboard.putNumber("LimelightX", x);
+		//SmartDashboard.putNumber("LimelightY", y);
+		//SmartDashboard.putNumber("LimelightArea", area);
+
+		//the above commented out code is for limelight
+								
+	public IntakeShooter() {}
+
+	public void setRPMIntake(){
+		//Intake
+	}
+    	
+    public void setRPMSpeaker(){
+		//method that gets the speaker motors up to speed
+	}
+
+	public void setRPMAmp(){
+		//method that gets the Shooter motors up to a speed to eject a note
+	}
+
+	public void setRPMEject(){
+		//method that runs the shooting and intake motors backwards
+	}
```

Comment:
These methods are rather limited. If you want the ability to shoot anywhere from the field, you will likely want to be able to set the outtake motors to any desired speed. Not one set RPM will work for all. Instead of making a new method for all the different RPMs, what should you do instead?